From 1cc6f2c342c1d3d1b49a07d20af5d33076c8760d Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Sat, 22 Apr 2023 22:22:34 +0300
Subject: [PATCH 28/28] Meson: Stop using deprecated get_cross_property()

Use get_external_property() instead.
This has also the benefit of adding support for
default values. One no longer need to give all
freeciv's properties, including ones completely unrelated
to the build in progress, in custon cross files.

See osdn #44913

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index d58036002e..5392a59d8f 100644
--- a/meson.build
+++ b/meson.build
@@ -155,10 +155,10 @@ else
 endif
 
 if meson.is_cross_build()
-  cross_inc_str = meson.get_cross_property('cross_inc_path')
+  cross_inc_str = meson.get_external_property('cross_inc_path', '')
   cross_inc_path = [cross_inc_str]
-  cross_lib_path = [meson.get_cross_property('cross_lib_path')]
-  crosser = meson.get_cross_property('crosser')
+  cross_lib_path = [meson.get_external_property('cross_lib_path', '')]
+  crosser = meson.get_external_property('crosser', false)
 
   # emscripten build is always cross-build, so check it only here
   if c_compiler.compiles(
-- 
2.39.2

